akka-actor-typed/src/main/scala/akka/actor/typed/internal/routing/PoolRouterImpl.scala

```diff
@@ -39,12 +39,12 @@ private[akka] final case class PoolRouterBuilder[T](
 private final class PoolRouterImpl[T](
     ctx: ActorContext[T],
     poolSize: Int,
-    behavior: Behavior[T],
+    behaviorFactory: () => Behavior[T],
     logic: RoutingLogic[T])
     extends AbstractBehavior[T] {
 
   (1 to poolSize).foreach { _ =>
-    val child = ctx.spawnAnonymous(behavior)
+    val child = ctx.spawnAnonymous(behaviorFactory())
```

Comment:
behavior factory adapter context wrapper

Reply:
At least it's not abstract